ur_robot_driver/src/hardware_interface.cpp

```diff
@@ -374,35 +376,26 @@ return_type URPositionHardwareInterface::write()
        runtime_state_ == static_cast<uint32_t>(rtde::RUNTIME_STATE::PAUSING)) &&
       robot_program_running_ && (!non_blocking_read_ || packet_read_))
   {
-    RCLCPP_INFO(rclcpp::get_logger("URPositionHardwareInterface"), "Writing ...");
-
-    // create a lambda substract functor
-    std::function<double(double, double)> substractor = [](double a, double b) { return std::abs(a - b); };
-
-    // create a position difference vector
-    std::vector<double> pos_diff;
-    pos_diff.resize(urcl_position_commands_.size());
-    std::transform(urcl_position_commands_.begin(), urcl_position_commands_.end(), urcl_position_commands_old_.begin(),
-                   pos_diff.begin(), substractor);
+    if (!position_interface_in_use_)
```

Comment:
With this logic here you will not call `writeJointCommand` in the same update call when you detect `position_interface_in_use_ = true;`

Is this correct? The old velocity implementation will do this in the same loop.

Reply:
Correct!
https://github.com/PickNikRobotics/Universal_Robots_ROS2_Driver/pull/32/commits/4aad52397470f02486bd9bc269fbb368361dad00